geozero/src/gdal/gdal_reader.rs

```diff
@@ -147,12 +195,71 @@ fn process_polygon<P: GeomProcessor>(
     processor.polygon_end(tagged, idx)
 }
 
+fn process_triangle<P: GeomProcessor>(
+    geo: &Geometry,
+    tagged: bool,
+    idx: usize,
+    processor: &mut P,
+) -> Result<()> {
+    let ring_count = geo.geometry_count();
+    processor.triangle_begin(tagged, ring_count, idx)?;
+    process_linestring_seq(geo, processor, ring_count)?;
+    processor.triangle_end(tagged, idx)
+}
+
+fn process_compoundcurve<P: GeomProcessor>(
+    geo: &Geometry,
+    idx: usize,
+    processor: &mut P,
+) -> Result<()> {
+    let curve_count = geo.geometry_count();
+    processor.compoundcurve_begin(curve_count, idx)?;
+    for i in 0..curve_count {
+        let geom = unsafe { geo.get_unowned_geometry(i) };
+        let ty: OGRwkbGeometryType::Type = type2d(geom.geometry_type());
+        match ty {
+            OGRwkbGeometryType::wkbLineString => {
+                process_linestring(&geom, false, i, processor)?;
+            }
+            OGRwkbGeometryType::wkbCircularString => {
+                process_circularstring(&geom, i, processor)?;
+            }
+            _ => return Err(GeozeroError::GeometryFormat),
+        }
+    }
+    processor.compoundcurve_end(idx)
+}
+
+fn process_curve<P: GeomProcessor>(geo: &Geometry, idx: usize, processor: &mut P) -> Result<()> {
+    let ty: OGRwkbGeometryType::Type = type2d(geo.geometry_type());
+    match ty {
+        OGRwkbGeometryType::wkbLineString => process_linestring(geo, false, idx, processor),
+        OGRwkbGeometryType::wkbCircularString => process_circularstring(geo, idx, processor),
+        OGRwkbGeometryType::wkbCompoundCurve => process_compoundcurve(geo, idx, processor),
+        _ => Err(GeozeroError::GeometryFormat),
+    }
+}
+
+fn process_curvepolygon<P: GeomProcessor>(
+    geo: &Geometry,
+    idx: usize,
+    processor: &mut P,
+) -> Result<()> {
+    let ring_count = geo.geometry_count();
+    processor.curvepolygon_begin(ring_count, idx)?;
+    for i in 0..ring_count {
+        let geom = unsafe { geo.get_unowned_geometry(i) };
+        process_curve(&geom, i, processor)?;
+    }
+    processor.curvepolygon_end(idx)
+}
+
 #[cfg(test)]
 #[cfg(feature = "with-wkt")]
```

Comment:
We need a feature test for `with-wkb` here.

Reply:
Done 👍